common-ui-compose/src/main/java/app/tivi/common/compose/Insets.kt

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.tivi.common.compose
+
+import androidx.compose.Composable
+import androidx.compose.Providers
+import androidx.compose.StructurallyEqual
+import androidx.compose.getValue
+import androidx.compose.mutableStateOf
+import androidx.compose.onCommit
+import androidx.compose.remember
+import androidx.compose.setValue
+import androidx.compose.staticAmbientOf
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.ui.core.DensityAmbient
+import androidx.ui.core.Modifier
+import androidx.ui.core.ViewAmbient
+import androidx.ui.core.composed
+import androidx.ui.layout.absolutePadding
+import androidx.ui.layout.preferredHeight
+import androidx.ui.layout.preferredWidth
+import androidx.ui.unit.Density
+import androidx.ui.unit.Dp
+import androidx.ui.unit.dp
+
+/**
+ * Main holder of our inset values.
+ *
+ * TODO add other inset types (IME, visibility, etc)
+ */
+class DisplayInsets {
```

Comment:
Rather than having this holder class, what about just making each property of `Insets` a `mutableStateOf` and then updating each?

Reply:
Good idea, done.

Reply:
Do we need the `DisplayInsets` class? Can't we just hold an `Insets` instance in the ambient?

Reply:
Yep, eventually I'll add `IME` insets. I'll add the gesture insets now.